CORE-3605 Fix error when creating an assessment

goody44 opened the pull request:
The program relationship was being created multiple times and failing a uniqueness constraint on the table.

Smotko commented on src/ggrc/models/hooks/assessment.py:
```diff
@@ -30,7 +30,6 @@ def handle_assessment_post(sender, obj=None, src=None, service=None):
 
     map_assessment(obj, src.get("object"))
     map_assessment(obj, src.get("audit"))
-    map_assessment(obj, src.get("program"))
```
If you remove this line, the program will not get mapped when we create the assessment through our generate assessment modal. Have you figured out where the other call for creating a relationship is coming from? It might be better if we delete that one.

goody44 commented:
@Smotko The duplicate is coming from the call I removed. The program is getting mapped in the "map_assessment(obj, src.get("object"))" call. And then it's getting mapped again in the line I removed.

Smotko commented:
@goody44 Ah, I see. I thought we the other mapping was done from the frontend. In this case we need to create the program mapping only if `src.get("object")` is a different object than `src.get("program")`.

edofic commented on src/ggrc/models/hooks/assessment.py:
```diff
@@ -28,9 +28,15 @@ def handle_assessment_post(sender, obj=None, src=None, service=None):
+    map_assessment(obj, audit)
+    # The program may also be set as the src_obj. If so then it should not be
+    # mapped again.
+    if src_obj["id"] != program["id"] or src_obj["type"] != program["type"]:
```
Both `src_obj` and `program` can be `None` here. This is why integration tests fail.

edofic commented:
:+1:
